### ISSUE TITLE
false alarm for sign-off

### ISSUE THREAD

**Issue by danielnaber** [languagetool-org/languagetool#2363]
This text triggers `PUNCTUATION_PARAGRAPH_END`:

```
Hi,

this is a test.

Best Regards,
Daniel
```

... while the same text with `Regards` instead of `Best Regards` works fine. @FredKruse I see where this happens in the code (`isFirstWord = (isWord(tokens[1]) && !isPunctuationMark(tokens[2]))`) but do you have an elegant idea how to fix this? Just extend the `if`? Note there are also sign-offs with three words like "all the best".

**Comment by FredKruse** [comment]
@danielnaber This bug is difficult. In the example above, the standalone version assumes only two paragraphs. It should be three, I think. If you translate the text into German there are three paragraphs.
In the example above the first paragraph is only "Hi,". The second includes the rest of the text. So the rule assumes two sentences. This generates the false alarm.
I am not really sure but I think the problem regards the sentences tokenizier. Shouldn't the text to be splitted in three paragraphs?

**Comment by danielnaber** [comment]
> Shouldn't the text to be splitted in three paragraphs?

I'm not sure. We have two modes: `\n` starts a new paragraph or `\n\n` starts a new paragraph. But I'm not sure which one is "correct". But let me try to find another workaround for this issue.

**Comment by FredKruse** [comment]
As far as I know \n\n is the default. The mode \n has to be set explicitly. But if \n is set, it should be more than 3 paragraphs. So anything is wrong.

**Comment by danielnaber** [comment]
> So anything is wrong.

How exactly did you test? I tried both German and English with the `-v` flag for the command line LT, and I didn't see a difference in sentence splitting for both (using the example text from above).

**Comment by FredKruse** [comment]
For test reasons I built in some debug information direct into the rule.
I wonder if the \n\n are positioned at the beginning of next sentence instead of at the end of the last sentence of paragraph. In this case the rule would not recognize the end of paragraph. The rule uses the method hasParagraphEndMark(Language lang) at class AnalyzedSentence to detect the end of a paragraph.
I will test it in the next days.

**Comment by FredKruse** [comment]
I did another test. The two \n after the first row are stored at the end of the sentence ("Hi," is handled as a sentence). After that only one \n is stored in the last sentence of a paragraph. The second \n is removed. This is happens for English texts. In German texts two \n are stored at the end of the last sentence of a paragraph.
So paragraph handling rules have no chance to recognize the end of a paragraph. Is there a possibility to change that?

**Comment by danielnaber** [comment]
> In German texts two \n are stored at the end of the last sentence of a paragraph.

How do you test this? When I just print out `sentence.getText()`, the analysis looks the same to me wrt. whitespace?

**Comment by FredKruse** [comment]
I printed out sentence.getText() too. The problem is shown when you add some character after the sentence, (eg. sentence.getText() + "#"). This shows that sometimes both \n are added to the last sentence of a paragraph and sometimes one \n is added to the end of the last sentence and one \n at the beginning of the next sentence.
I wonder if this is feature or a bug. For German language the \n\n seems to be always at the end of the last sentence of paragraph. I changed the method for testing of end of paragraph and changed it in all affected rules. So it should work also for English language now.
